core/meta/src/TCling.cxx

```diff
@@ -7343,6 +7343,14 @@ void TCling::GetFunctionName(const clang::FunctionDecl *decl, std::string &outpu
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+
+void TCling::NullDerefDiagnose(std::exception e) const
+{
+   if (dyn_cast<InterpreterException>(e))
+      e.diagnose();
```

Comment:
That shouldn't even compile :-) `e` is still a `std::exception` and does not have `diagnose()`.

Here is what I'd prefer: add `diagnose()` as virtual function in `InterpreterException`, then

```
if (auto ie = dyn_cast<InterpreterException>(&e)) {
   ie->diagnose();
}
```

Then we can rename `NullRederDiagnose()` to `DiagnoseIfInterpreterException()`. This function will get called about 2 times during the lifetime of LHC ;-) so a long, expressive name is just fine with me.